src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -0,0 +1,60 @@
+use crate::commands::beta::generate::utils::Language;
+use serde::Serialize;
+use tera::Tera;
+
+/// Represents a context that will be used for code generation
+pub struct CodeGenContext {
+    // Represents the templating engine - tera
+    pub(crate) tera: Tera,
+    pub(crate) language: Language,
+    // Initially the data_model field is set to None.
+    // Once an ISL type definition is mapped to a data model this will have Some value.
+    pub(crate) data_model: Option<DataModel>,
+    // Represents a counter for naming anonymous type definitions
+    pub(crate) anonymous_type_counter: usize,
+}
+
+impl CodeGenContext {
+    pub fn new(language: Language) -> Self {
+        Self {
+            language,
+            data_model: None,
+            anonymous_type_counter: 0,
+            tera: Tera::new("src/bin/ion/commands/beta/generate/templates/**/*.templ").unwrap(),
+        }
+    }
+
+    pub fn with_data_model(&mut self, data_model: DataModel) {
+        self.data_model = Some(data_model);
+    }
+
+    pub fn with_initial_data_model(&mut self) {
+        // Initially the data model is set to None, this will be set with Some(_) value when data model is determined in code generation process
+        self.data_model = None;
+    }
+
+    /// Returns a string that represent the template name based on data model type.
+    pub fn template_name(&self) -> &str {
```

Comment:
Template name refers to the name of the template file which will be used to generate code. (e.g. struct.templ)
It can be defined as an enum but like you said would need to have `fn name(&self) -> &str`.

Reply:
Ah, ok! I didn't get that this was for the file name. Yeah, I think having an enum of the templates we support (maybe broken out by language?) with a `name` method would make sense.